### PR TITLE
chore: add `deduplicate-yarn` workflow

### PR DESCRIPTION
**Pull request by mcansh**
automation of #6108

Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

File: .github/workflows/dedupe-yarn.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: 👔 Deduple yarn.lock
```
Comment (MichaelDeBoey) [inline .github/workflows/dedupe-yarn.yml:1]:
```suggestion
name: 👔 Deduplicate yarn.lock
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: .github/workflows/dedupe-yarn.yml (hunk 1 above)
Comment (mcansh) [reply .github/workflows/dedupe-yarn.yml]: quickly done on my phone 😅

File: .github/workflows/dedupe-yarn.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 🛑 Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.11.0
+
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.FORMAT_PAT }}
```
Comment (MichaelDeBoey) [inline .github/workflows/dedupe-yarn.yml:23]: We don't need to pass `token` if we set write permissions (like we do in `no-response`) I think 🤔

**Review by MichaelDeBoey — APPROVED** [review]
LGTM! 😍

#6108 still needs to be merged though

**Comment by mcansh** [comment]
> LGTM! 😍
> 
> 
> 
> #6108 still needs to be merged though

i'll check it out in the morning, didnt want to just yolo approve from my phone on this one

File: .github/workflows/dedupe-yarn.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+on:
+  push:
+    branches:
+      - main
+      - dev
+    paths:
+      - ./yarn.lock
+
+jobs:
```
Comment (MichaelDeBoey) [inline .github/workflows/dedupe-yarn.yml:11]:
```suggestion
concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

jobs:
```

File: .github/workflows/dedupe-yarn.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  format:
+    if: github.repository == 'remix-run/remix'
+    runs-on: ubuntu-latest
+
+    permissions:
+      # Give the default GITHUB_TOKEN write permission to commit and push the
+      # added or changed files to the repository.
+      contents: write
+
```
Comment (MichaelDeBoey) [inline .github/workflows/dedupe-yarn.yml:20]:
```suggestion
```

File: .github/workflows/dedupe-yarn.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    steps:
+      - name: 🛑 Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.11.0
+
```
Comment (MichaelDeBoey) [inline .github/workflows/dedupe-yarn.yml:24]:
```suggestion
```

File: .github/workflows/dedupe-yarn.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v3
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+          cache: "yarn"
+
+      - name: 👔 Dedupe yarn.lock
```
Comment (MichaelDeBoey) [inline .github/workflows/dedupe-yarn.yml:34]:
```suggestion
      - name: ️️⚙️ Dedupe yarn.lock
```

File: .github/workflows/dedupe-yarn.yml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        run: npx yarn-deduplicate && rm -rf ./node_modules && yarn
+
+      - name: 💪 Commit
+        run: |
+          git config --local user.email "hello@remix.run"
+          git config --local user.name "Remix Run Bot"
```
Comment (MichaelDeBoey) [inline .github/workflows/dedupe-yarn.yml:40]:
```suggestion
          git config --local user.email "github-actions[bot]@users.noreply.github.com"
          git config --local user.name "github-actions[bot]"
```

File: .github/workflows/dedupe-yarn.yml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+          git add .
+          if [ -z "$(git status --porcelain)" ]; then
+            echo "💿 no formatting changed"
```
Comment (MichaelDeBoey) [inline .github/workflows/dedupe-yarn.yml:44]:
```suggestion
            echo "💿 no deduplication needed"
```

File: .github/workflows/dedupe-yarn.yml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+            exit 0
+          fi
+          git commit -m "chore: dedupe yarn.lock"
```
Comment (MichaelDeBoey) [inline .github/workflows/dedupe-yarn.yml:47]:
```suggestion
          git commit -m "chore: deduplicate `yarn.lock`"
```

File: .github/workflows/format.yml [hunk 10]
```diff
@@ -11,14 +11,17 @@ jobs:
     if: github.repository == 'remix-run/remix'
     runs-on: ubuntu-latest
 
+    permissions:
+      # Give the default GITHUB_TOKEN write permission to commit and push the
+      # added or changed files to the repository.
+      contents: write
+
```
Comment (MichaelDeBoey) [inline .github/workflows/format.yml:18]:
```suggestion
```

File: .github/workflows/format.yml [hunk 11]
```diff
@@ -11,14 +11,17 @@ jobs:
...
     steps:
       - name: 🛑 Cancel Previous Runs
         uses: styfle/cancel-workflow-action@0.11.0
 
       - name: ⬇️ Checkout repo
         uses: actions/checkout@v3
```
Comment (MichaelDeBoey) [inline .github/workflows/format.yml:24]:
```suggestion
        uses: actions/checkout@v3
        with:
          token: ${{ secrets.FORMAT_PAT }}
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]
Let's keep this consistent with remix-run/react-router#10583 and remix-run/examples#248 & remix-run/examples#263

**Review by MichaelDeBoey — APPROVED** [review]

**Review by brophdawg11 — APPROVED** [review]

**Pull request merged**